src/Sulu/Component/Content/Compat/Block/BlockProperty.php

```diff
@@ -85,6 +87,17 @@ public function doSetValue($value)
             return;
         }
 
+        if (!\is_array($items)) {
+            if (\is_object($items)) {
+                $itemsString = 'object of class ' . \get_class($items);
+            } else {
+                $itemsString = \sprintf('"%s"', \var_export($items, true));
+            }
+            throw new \InvalidArgumentException(
+                \sprintf('Expected block configuration but got %s at property: %s', $itemsString, $this->getName())
+            );
```

Comment:
We should keep this simple:

```suggestion
            throw new \InvalidArgumentException(
                \sprintf('Expected block configuration but got "%s" at property: "%s"', \get_debug_type($itemsString), $this->getName())
            );
```

Reply:
Does this also work in production since this is a debug function.

Reply:
Yes it is also available on prod php version aslong it is PHP 8.0. But as this is more an improvement and not a bugfix its fine to be merged this way into the 2.5 branch.

Reply:
Ah a new php 8 feature. Good to know.